dask/order.py

```diff
@@ -683,12 +684,27 @@ class StrComparable(object):
     """
 
     __slots__ = ("obj",)
+    zfill_pattern = re.compile(r", \d+")
+
+    @staticmethod
+    def zerofill_key(match):
+        """Zero fill integers in keys of the form: `(<str>, <int>, <int>, ...)`
+
+        HACK: we zerofill 9 decimals, which should be enough in all but extreme cases.
+        """
+        value = str(match.group())[2:]
+        return value.zfill(9)
```

Comment:
This seems like it might be expensive.  I recommend that we wait on this for a bit.